### ISSUE TITLE
retry_on_initial_connect

### ISSUE THREAD

**Issue by stevelr** [nats-io/nats.rs#711]
1 bug and 1 suggestion:

(1) `fn retry_on_intial_connect` is misspelled 

(2) This implementation 
```
    let duration = if *server_attempts == 0 {
          Duration::from_millis(0)
      } else {
          let exp: u32 = (*server_attempts - 1).try_into().unwrap_or(std::u32::MAX);
          let max = Duration::from_secs(4);
          cmp::min(Duration::from_millis(2_u64.saturating_pow(exp)), max)
      };
     *server_attempts += 1;
```

could be simplified to the following with nearly the same effect:
```
    let duration = Duration::from_millis(cmp.min(*server_attempts, 4) as u64);
    *server_attempts += 1;
```
The first generates the sequence [0,1,2,4,4,...] and the second generates [0,1,2,3,4,4,...]

**Comment by stevelr** [comment]
If anybody uses the misspelled version in 0.22 and you fix the spelling in a future release, it would be a breaking change. To avoid the breaking change, the fix release would need to include the fixed spelling, and that release plus the next few would also include the current misspelled method, with a deprecation flag.

Since 0.22.0 has only been out for a day, you might be able to get away with 0.22.1 and yanking 0.22.0; thereby avoiding having to include the misspelling in the next few releases to retain backwards compatibility.

**Comment by Jarema** [comment]
fixed in #713 

Release will be issuesd today.

**Comment by Jarema** [comment]
@stevelr the 0.22.0 has been yanked.
